backend/app/schemas/dataset_item.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from enum import StrEnum
+from uuid import UUID
+
+from pydantic import BaseModel
+
+from app.schemas.base import BaseIDNameModel, Pagination
+
+
+class DatasetItemFormat(StrEnum):
+    JPG = "jpg"
+    PNG = "png"
+
+
+class DatasetItemSubset(StrEnum):
+    UNASSIGNED = "unassigned"
+
+
+class DatasetItem(BaseIDNameModel):
+    """
+    Dataset item
+    """
+
+    format: DatasetItemFormat
+    width: int
+    height: int
+    size: int
+    source_id: UUID
```

Comment:
```suggestion
    source_id: UUID | None
```
This one is nullable, for items that are uploaded by the user instead of acquired from the pipeline